modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,241 @@ public boolean updateResource(String applicationKey, String resourceType, String
         return false;
     }
 
+    /**
+     * Add deployement config details
+     *
+     * @param deploymentConfig deployement config
+     */
+    public void addDeploymentConfig(DeploymentConfig deploymentConfig) throws AppFactoryException {
+        Connection connection = null;
+        PreparedStatement addDeploymentPreparedStatement = null;
+
+        try {
+            connection = AppFactoryDBUtil.getConnection();
+            addDeploymentPreparedStatement = connection.prepareStatement(SQLConstants.ADD_DEPLOYMENT);
+            addDeploymentPreparedStatement.setString(1, deploymentConfig.getDeploymentName());
+            addDeploymentPreparedStatement.setInt(2, deploymentConfig.getReplicas());
+            addDeploymentPreparedStatement.executeUpdate();
+
+            List<Container> containerList = deploymentConfig.getContainers();
+            addContainer(containerList, deploymentConfig.getDeploymentName(), connection);
+            if(addDeploymentPreparedStatement.getUpdateCount() > 0){
+                connection.commit();
+            }
+
+        } catch (SQLException e) {
+            try {
+                if (connection != null) {
+                    connection.rollback();
+                }
+            } catch (SQLException rollbackExcpetion) {
+                log.error("Error while rolling back the added application", rollbackExcpetion);
+            }
+            String message = "Error while adding database for deployement config : "
+                    + deploymentConfig.getDeploymentName();
+            log.error(message, e);
+            throw new AppFactoryException(message, e);
+        } finally {
+            AppFactoryDBUtil.closePreparedStatement(addDeploymentPreparedStatement);
+            AppFactoryDBUtil.closeConnection(connection);
+        }
+    }
+
+    /**
+     * Get deployement config
+     *
+     * @param applicationId id of the application
+     * @param stage         stage of the application
+     * @return deployement config
+     */
+    public DeploymentConfig getDeploymentConfig(String applicationId, String stage) throws AppFactoryException {
+        Connection databaseConnection = null;
+        PreparedStatement getDeploymentPreparedStatement = null;
+        ResultSet deployementResultSet = null;
+
+        DeploymentConfig deploymentConfig = new DeploymentConfig();
+
+        try {
+            databaseConnection = AppFactoryDBUtil.getConnection();
+            getDeploymentPreparedStatement = databaseConnection.prepareStatement(SQLConstants.GET_DEPLOYEMENET);
+
+            String deploymentName = applicationId + "-" + stage;
```

Comment:
This is fixed in following commit.
https://github.com/wso2/product-af/pull/800/commits/b54500739fe65b9908eda6176ecbace4b76a972d